index.js

```diff
@@ -158,39 +159,48 @@ function _make_generateLog(metrics, func, start_time, config, context) {
           log('IOPIPE-DEBUG: ', JSON.stringify(response_body))
         }
 
-        var req = https.request({
-          hostname: config.url,
-          path: config.path,
-          port: 443,
-          method: 'POST',
-          headers: {'content-type' : 'application/json'},
-          agent: httpsAgent,
-          timeout: config.networkTimeout
-        }, (res) => {
-          var apiResponse = ''
-
-          res.on('data', function (chunk) {
-            apiResponse += chunk
-          })
+        dnsPromise.then((ipAddress) => {
+          var req = https.request({
+            hostname: ipAddress,
```

Comment:
the dnsPromise is assigned/created per-invocation so if the dns cache is expired, a new DNS query will be made, and a new IP address can be grabbed in the event of an IP change (for maintenance, scaling, etc reasons)